app/src/main/java/edu/mills/heartoftaiwanese/activity/adapter/WordListAdapter.kt

```diff
@@ -0,0 +1,125 @@
+package edu.mills.heartoftaiwanese.activity.adapter
+
+import android.content.res.ColorStateList
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import edu.mills.heartoftaiwanese.R
+import edu.mills.heartoftaiwanese.data.DatabaseWord
+import edu.mills.heartoftaiwanese.data.Word
+import edu.mills.heartoftaiwanese.databinding.LayoutWordItemBinding
+import java.text.DateFormat
+
+class WordListAdapter :
+    ListAdapter<DatabaseWord, WordListAdapter.WordListViewHolder>(WordsCallback) {
+
+    var favoriteButtonListener: FavoriteButtonListener? = null
+
+    interface FavoriteButtonListener {
+        /**
+         * Interface function that determines what can happen if the favorite is clicked.
+         * The view adapter takes care of hiding/showing the favorite button.
```

Comment:
I'm not sure what you mean by this. Do you mean that the caller of this function doesn't need to worry about it?